qml/js/yleApi.js

```diff
@@ -28,12 +28,57 @@ var filterAvailablePrograms = function(programs) {
     }, []);
 }
 
+var zeropad = function(str, size) {
+    var s = str;
+    while (s.length < size) s = "0" + s;
+    return s;
+}
+
+var parseDuration = function(dur) {//PT1H58M45S
+    if (!dur)
+        return "";
+    var d = dur.substr(2);
+    var hi = d.indexOf("H");
+    var mi = d.indexOf("M");
+    var si = d.indexOf("S");
+    var h = d.substr(0, hi);
+    var m = zeropad(d.substring(hi+1, mi), 2);
+    var s = zeropad(d.substr(mi+1, si >= 0 ? si-mi-1 : si).substr(0, 2), 2);
+    return h ? ""+h+":"+m+":"+s : ""+m+":"+s;
+}
+
+var parseTime = function(str) {//2014-01-23T21:00:07+02:00
+    var s = Qt.formatDateTime(new Date(str), "ddd dd.MM hh:mm")
+    console.log(str+"->"+s)
+    return s;
+}
+
+function arraySearch(arr, key, sub, val) {
+    for (var i = 0; i < arr.count; i++) {
+        if (sub && arr.get(i)[sub] && arr.get(i)[sub][key] === val)

```

Comment:
Where are `.get` and `.count` functions coming from?

Reply:
Those should be basic javascript array functions.

Reply:
I might be confused also but I'm pretty sure they aren't, they might result from Array / ListModel conversions, but I did not see any explicit conversions happening anywhere.

Should check with typeof & console logs what kind of object it actually is :)